src/Shared/AssemblyVersionExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using System.Reflection;
+
+namespace OpenTelemetry.Instrumentation;
```

Comment:
nit: @Kielek If you do a follow-up PR to make changes to OtlpExporter I think we should switch this to `OpenTelemetry.Internal` namespace. It just seems a bit weird seeing `using OpenTelemetry.Instrumentation;` in the SDK.

Reply:
+1